WallsAndHoles/xmltool.h

```diff
@@ -0,0 +1,19 @@
+#ifndef XMLTOOL_H
+#define XMLTOOL_H
+#include "tilemap.h"
+#include "tiletemplateset.h"
+#include <QDomDocument>
+#include <QFile>
+#include <QTextStream>
+#include <QXmlStreamReader>
+
+namespace XMLTool {
+
+    SharedTileMap openTileMap(QString tileMapPath);
+    SharedTileTemplateSet openTileTemplateSet(QString templateSetPath);
+
+    int saveTileMap(SharedTileMap tileMap, bool saveTemplates = true);
```

Comment:
remove default for saveTemplates. This should be explicitly stated to avoid unwanted behavior.

Reply:
Seemly if you save tilemap without save its depending templates, it may end up holding a file path pointing to nowhere, and thus can not be reload...unless you manually save template to that path.